tools/license/get_dep_licenses.go

```diff
@@ -184,12 +254,61 @@ func main() {
 		linfo.path = lp
 		linfo.url = pathToURL(lp)
 		licenses = append(licenses, linfo)
+		ltypeStr := linfo.licenseTypeString
 		if linfo.exact {
-			licenseTypes[linfo.licenseTypeString] = append(licenseTypes[linfo.licenseTypeString], p)
+			licenseTypes[ltypeStr] = append(licenseTypes[ltypeStr], p)
 			exact = append(exact, linfo)
 		} else {
 			inexact = append(inexact, linfo)
 		}
+
+		switch {
+		case approvedLicenses[ltypeStr]:
+		case reciprocalLicenses[ltypeStr]:
+			reciprocalList = append(reciprocalList, linfo.packageName)
+		case restrictedLicenses[ltypeStr]:
+			restrictedList = append(restrictedList, linfo.packageName)
+		case ltypeStr == "":
+			missingList = append(missingList, linfo.packageName)
+		default:
+			if !knownUnknownLicenses[linfo.packageName] {
+				unknownMap[linfo.packageName] = ltypeStr
+			}
+		}
+	}
+
+	if check {
```

Comment:
perhaps move if check to before switch above (so these lists are not appended).?

Reply:
I think you mean to add the same check inside the for loop, around the switch?
It's not a performance critical program so I'd err on the side of trimming unnecessary indentation to improve readability.